main.c

```diff
@@ -137,7 +142,68 @@ static void linux_boot(const struct boot_params boot_params)
     printf("  pref_address: 0x%llx\n", l_params->pref_address);
     printf("\n");
 
-    char *linux_header_end = (char *)(boot_params.kernel_addr + 0x201);
+    // Checks if a relocation of the kernel is required.
+    // We decided that the loader moves all kernels to their preferred address, even if they
+    // are relocatable, thus, could relocate themselves.
+    if (boot_params.kernel_addr == l_params->pref_address)
+    {
+        printf(
+            "The kernel is already loaded at its preferred address 0x%llx\n",
+            l_params->pref_address
+        );
+    } else {
+        printf(
+            "The kernel will be moved from 0x%lx to its preferred load address 0x%llx.\n",
```

Comment:
nit: why so much text? What's wrong with:

```suggestion
            "Relocating kernel from 0x%lx to 0x%llx.\n",
```